File: backend/platform/database/location.go
package database

import "going-going-backend/platform/database/array"

type Location struct {
	Id         *uint64          `gorm:"primaryKey" json:"id"`
	Name       *string          `gorm:"type:VARCHAR(255); not null" json:"name"`
	Address    *string          `gorm:"type:VARCHAR(255); not null" json:"address"`
	Coordinate array.Coordinate `gorm:"type:string;" json:"coordinate"`
}

File: frontend/lib/models/location.dart
class Location {
  int? id;
  double lat;
  double lng;
  String name;
  String address;

  Location({
    this.id,
    required this.lat,
    required this.lng,
    this.name = "",
    this.address = "",
  });

  Location.fromJson(Map<String, dynamic> json)
      : id = json["id"],
        lat = json["coordinate"][0],
        lng = json["coordinate"][1],
        name = json["name"],
        address = json["address"];

  Map<String, dynamic> toJson() => {
        "id": id,
        "coordinate": [lat, lng],
        "name": name,
        "address": address,
      };
}
